code/modules/mob/living/death.dm

```diff
@@ -104,6 +104,12 @@
 		if(SSlag_switch.measures[DISABLE_DEAD_KEYLOOP] && !client?.holder)
 			to_chat(src, span_deadsay(span_big("Observer freelook is disabled.\nPlease use Orbit, Teleport, and Jump to look around.")))
 			ghostize(TRUE)
+	if (SSgamemode.current_storyteller && SSgamemode.current_storyteller.type == /datum/storyteller/starved)
```

Comment:
I suggest looking into the way that the deathless final ritual handles things, as I think it uses signals

Reply:
oh yeah, `COMSIG_GLOB_MOB_DEATH`